api/accesslog.proto

```diff
@@ -0,0 +1,138 @@
+    // Request timeout on upstream.
+    UPSTREAM_REQUEST_TIMEOUT = 2;
+    // Local codec level reset was sent on the stream.
+    LOCAL_RESET = 3;
+    // Remote codec level reset was received on the stream.
+    UPSTREAM_REMOTE_RESET = 4;
+    // Local reset by a connection pool due to an initial connection failure.
+    UPSTREAM_CONNECTION_FAILURE = 5;
+    // If the stream was locally reset due to connection termination.
+    UPSTREAM_CONNECTION_TERMINATION = 6;
+    // The stream was reset because of a resource overflow.
+    UPSTREAM_OVERFLOW = 7;
+    // No route found for a given request.
+    NO_ROUTE_FOUND = 8;
+    // Request was delayed before proxying.
+    DELAY_INJECTED = 9;
+    // Abort with error code was injected.
+    FAULT_INJECTED = 10;
+    // Request was ratelimited locally by rate limit filter.
+    RATE_LIMITED = 11;
+  }
+  // Status flags about the response.
+  repeated ResponseFlag response_flags = 4;
+
+  // The remote (upstream) host, as an IPv4 or IPv6 string.
+  // IPv6 addresses should be stored in canonical (compressed) format.
+  string upstream_host = 5;
+
+  // The Upstream Cluster that the upstream host belongs to.
+  string upstream_cluster = 6;
+
+  // This field is the IP and port on which the request from the user was
+  // received, stored in ipv4:port or [ipv6]:port format.
+  string destination_host = 7;
+
+  // Size of the HTTP request body in bytes
+  google.protobuf.UInt64Value request_body_size = 8;
+
+  // Size of the HTTP response body in bytes
+  google.protobuf.UInt64Value response_body_size = 9;
+
+  // Size of the HTTP request headers in bytes
+  google.protobuf.UInt64Value request_hdrs_size = 10;
+
+  // Size of the HTTP response headers in bytes
+  google.protobuf.UInt64Value response_hdrs_size = 11;
+
+  // Whether the request arrived via a secure (TLS) protocol
+  google.protobuf.BoolValue secure = 12;
+
+  // Whether the request is a HealthCheck request
+  google.protobuf.BoolValue health_check = 13;
+
+  // The HTTP response code
+  google.protobuf.UInt32Value response_code = 14;
+
+  // User agent as sent by client HTTP
+  string user_agent = 15;
+
+  // Path
+  string path = 17;
+
+  // Referer field as sent by client HTTP
+  // (Referer is spelled to match the HTTP spec, not English).
+  string referer = 18;
+
+  // X-Forwarded-For request header
+  string forwarded_for = 19;
+
+  // X-Request-Id request header
+  string request_id = 20;
+
+  // HTTP2 :authority header value or HTTP1.1 Host header value
+  string authority = 21;
+
+  // Duration (milliseconds)
+  //
+  // The total duration it took to service this request from the StartTime until
+  // the response was written to the user.
+  google.protobuf.Duration duration = 22;
+
+  // Upstream Service Time
+  //
+  // From the X-Envoy-Upstream-Service-Time response header, this is the amount it took
+  // the upstream server to service the request.
+  google.protobuf.Duration upstream_service_time = 23;
```

Comment:
This comes directly from the X-ENVOY-UPSTREAM-SERVICE-TIME response header so that's why "time" was used.  I agree that the word duration makes more sense - it's definitely a duration - so I think I will make this change.